src/controllers/dacpacDialogWebviewController.ts

```diff
@@ -689,104 +623,167 @@ export class DacFxApplicationWebviewController extends ReactWebviewPanelControll
-                };
             }
 
-            // Case 2: Connection exists but not connected - auto-connect
-            if (!matchingConnection.isConnected) {
-                this.logger.verbose(`Auto-connecting to profile: ${matchingConnection.profileId}`);
-                try {
-                    const connectResult = await this.connectToServer(matchingConnection.profileId);
-
-                    if (connectResult.isConnected && connectResult.ownerUri) {
-                        ownerUri = connectResult.ownerUri;
-                        updatedConnections = connections.map((conn) =>
-                            conn.profileId === matchingConnection.profileId
-                                ? { ...conn, isConnected: true }
-                                : conn,
-                        );
-                        this.logger.info(
-                            `Successfully auto-connected to: ${matchingConnection.server}`,
-                        );
-                        return {
-                            connections: updatedConnections,
-                            selectedConnection: { ...matchingConnection, isConnected: true },
-                            ownerUri,
-                            autoConnected: true,
-                        };
-                    } else {
-                        // Connection failed
-                        this.logger.error(
-                            `Auto-connect failed: ${connectResult.errorMessage || "Unknown error"}`,
-                        );
-                        return {
-                            connections,
-                            selectedConnection: matchingConnection,
-                            autoConnected: false,
-                            errorMessage: connectResult.errorMessage,
-                        };
-                    }
-                } catch (error) {
-                    const errorMsg = error instanceof Error ? error.message : String(error);
-                    this.logger.error(`Auto-connect exception: ${errorMsg}`);
-                    return {
-                        connections,
-                        selectedConnection: matchingConnection,
-                        autoConnected: false,
-                        errorMessage: errorMsg,
-                    };
-                }
+            // Attempt to connect to the matched profile
+            return await this.connectToMatchedProfile(connections, matchingConnection);
+        } catch (error) {
+            this.logger.error(`Failed to initialize connection: ${error}`);
+            // Fallback: return empty state
+            return {
+                connections: [],
+                autoConnected: false,
+                errorMessage: error instanceof Error ? error.message : String(error),
+            };
+        }
+    }
+
+    /**
+     * Finds a matching connection profile based on profile ID or server/database name
+     */
+    private async findMatchingConnection(
```

Comment:
Can you make a github bug in the November milestone to clean this up further?  I'm going to make a small change to the matcher code that will handle all of this for you and eliminate the redundancy.